### PR TITLE
[docs] Update documentation for v1.0 release.

### LINKED ISSUE #48: Add "getting started" guide on how to set up a styleguide for your project

### LINKED ISSUE #167: Include Sass-JSON-export directly in Herman
Right now, in order to use many of the Herman features, you have to install the `sass-json-export` package, and include that in your own Sass along with Herman's export-enhancing utilities. We could simplify this by bundling simplified export tools in Herman.

Since sass-json export is pure CSS, this should be straightforward. And since Herman expects a particular output format, this would allow us to simplify the Sass API as well as the dependency management.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
Fixes #48.
Fixes #167.

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
Copied from #169:

A few subtle things that should be documented related to the `@font` annotation:

1) The font family name will default to the name of the font-map unless a `name` key is found.
2) If the font-map contains an `svgid` key, that value will be used for the `@font-face` `svg-file.svg#svgid` suffix. If none is found, it will default to the font family name.
3) If webfonts require additional HTML (e.g. external Google stylesheet or Typekit JS), they must be initialized with the `@font` annotation for them to work properly in `@example` annotations. The custom HTML defined in the `@font` annotation is automatically inserted into the `@example` annotation `<head>`.

File: scss/samples/_examples.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+/// ```yaml
+/// # .sassdocrc (yaml)
+/// herman:
+///   templatepath: 'templates'
+/// ```
+///
+/// ### Setting `nunjucksEnv`:
+///
+/// ```
+/// @@@
```
Comment (mirisuzanne) [inline scss/samples/_examples.scss]: @jgerigmeyer I'm not sure how to write an example of this…

**Comment by mirisuzanne** [comment]
@jgerigmeyer Ok, I don't know why I ended up merging sass-json-export into this branch, but I think we should merge, and improve the docs in smaller branches from here on out.

File: README.md [hunk 2]
```diff
@@ -1,119 +1,136 @@
-Herman [a SassDoc theme]
-========================
+# Herman [a SassDoc theme]
 
-[![Greenkeeper badge](https://badges.greenkeeper.io/oddbird/sassdoc-theme-herman.svg)](https://greenkeeper.io/)
```
Comment (jgerigmeyer) [inline README.md]: Was removing this intentional?

File: scss/utilities/_json-api.scss [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+///
+/// [json]: https://github.com/oddbird/sass-json-export
+///
+/// @group api_json-export
+/// @link https://github.com/oddbird/sass-json-export Sass-JSON-Export
+///
+/// @example scss
+///   @import 'sass-json-export/stylesheets/sass-json-export';
+///
+///   @include json-encode($herman, comment);
```
Comment (jgerigmeyer) [inline scss/utilities/_json-api.scss]: Is it confusing to include this as the first example, when we don't actually want users to use `sass-json-export` directly like this?

**Comment by jgerigmeyer** [comment]
@mirisuzanne I'm not clear on the goal here with including `sass-json-export`. Did you copy over all of the functionality (and tests) from https://github.com/oddbird/sass-json-export/, so it's no longer in use at all?

**Comment by mirisuzanne** [comment]
@jgerigmeyer All of the functionality and tests that are used by Herman, yes. That way Herman works without the extra dependency.

File: README.md [hunk 4]
```diff
@@ -1,121 +1,140 @@
...
+- Written directly in the code,
+  and integrated with code architecture
+- Automated for a document that grows and changes
+  along with the life of your project
 
+Herman is built as an extension to [SassDoc][sassdoc],
+and supports all their core functionality
+for font specimens, color palettes, sizes, SVG icons,
+compiled languages, Nunjucks/Jinja macros, HTML previews,
+and more.
```
[the remark this answers is not in the document]
Comment (mirisuzanne) [reply README.md]: yeah. that's a typo.

**Comment by jgerigmeyer** [comment]
I'm okay merging this as-is. I think the main thing missing is documentation for the custom annotations (`@font`, `@example`, `@preview`, and `@icons`).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+and everyone is welcome.
+As maintainers, we review all the code,
+and may provide feedback before accepting a PR.
+We're happy to work with you to make this the best
+(and friendliest) project we can.
+
+
+## Development
+
+To install the necessary Nodedependencies, run ``yarn``.
```
Comment (wlonk) [inline CONTRIBUTING.md]: `Node dependencies`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+– run ``gulp compile``.
+
+Just running ``gulp`` will perform all of the above tasks.
+
+You can start up a local development server with ``gulp serve``.
+This will also watch for changes to local files
+and automatically perform an appropriate selection of the above tasks
+whenever changes are detected to relevant files.
+
+Access the running server at http://localhost:3000.
```
Comment (wlonk) [inline CONTRIBUTING.md]: Put the URL in backticks?

File: README.md [hunk 7]
```diff
@@ -1,121 +1,222 @@
...
+
+All relative paths are relative to the SassDoc config file or the `cwd`.
+
+### displayColors
+
+- Type: `Array`
+- Default: `['hex', 'rgb', 'hsl']`
+
+Configures which color value formats are shown when using the
+[`@preview color-palette` annotation](http://oddbird.net/herman/docs/demo_colors.html).
```
Comment (wlonk) [inline README.md]: This URL 404s.

File: README.md [hunk 8]
```diff
@@ -1,121 +1,222 @@
...
+`String` path (in which case the filename will be displayed in the navigation).
+
+### customCSS
+
+- Type: `String`
+- Default: `''`
+
+Relative path to a custom CSS file,
+which will be included in the `<head>` of rendered
+[`@example` annotations](http://oddbird.net/herman/docs/demo_examples.html).
```
Comment (wlonk) [inline README.md]: This also 404s.

File: README.md [hunk 9]
```diff
@@ -1,121 +1,222 @@
...
+
+- Type: `String`
+- Default: `''`
+
+*Required if using `@icons` annotation, or if using `@example` annotations that
+require minified SVG icons.*
+
+Relative path to a file containing minified SVG icons:
+one `svg` element with many `symbol` elements,
+generated by https://github.com/Hiswe/gulp-svg-symbols or similar.
```
Comment (wlonk) [inline README.md]: Enlinken this.

File: README.md [hunk 10]
```diff
@@ -1,121 +1,222 @@
...
+in rendered `@icon` and `@example` annotations.
+
+### templatepath
+
+- Type: `String`
+- Default: `''`
+
+*Either `templatepath` or [`nunjucksEnv`](#nunjucksenv) is required if using
+`@icons` annotation or
+[`@example njk` annotation](http://oddbird.net/herman/docs/demo_examples.html#compiling-nunjucks).*
```
Comment (wlonk) [inline README.md]: Also 404s.

**Comment by jgerigmeyer** [comment]
The urls point to the eventual location of these docs. They will 404 until this branch is merged and deployed to OddSite.

**Comment by wlonk** [comment]
@jgerigmeyer ah, thanks. I didn't realize.

File: README.md (hunk 9 above)
Comment (jgerigmeyer) [reply README.md]: This is parsed as Markdown, so it gets auto-linkified.

File: README.md (hunk 2 above)
Comment (mirisuzanne) [reply README.md]: uh, if so I don't know why…

File: README.md [hunk 11]
```diff
@@ -1,121 +1,140 @@
...
+our `@icons` annotation allows you to
+display SVG icons from a given folder,
+and we provide a `@preview` command
+for displaying color-palettes, font-specimens,
+text and spacing sizes, and modular ratios.
+We also extend the core `@example` annotation
+to display compiled Sass/Nunjucks output
+and render sample components.
+
+[See the full documentation for details »][docs]
```
[the remark this answers is not in the document]
Comment (mirisuzanne) [reply README.md]: Yeah, I don't know the right way to go with that. Seems like a readme ought to link to the docs, but that can become recursive…

**Comment by mirisuzanne** [comment]
@jgerigmeyer I made a few adjustments if you want to review before merging.

File: README.md [hunk 12]
```diff
@@ -1,121 +1,103 @@
-Herman [a SassDoc theme]
-========================
+# Herman [a SassDoc theme]
 
 [![CircleCI](https://circleci.com/gh/oddbird/sassdoc-theme-herman.svg?style=svg)](https://circleci.com/gh/oddbird/sassdoc-theme-herman)
 [![Coverage Status](https://coveralls.io/repos/github/oddbird/sassdoc-theme-herman/badge.svg)](https://coveralls.io/github/oddbird/sassdoc-theme-herman)
 [![Greenkeeper badge](https://badges.greenkeeper.io/oddbird/sassdoc-theme-herman.svg)](https://greenkeeper.io/)
```
Comment (jgerigmeyer) [inline README.md]: Minor point, but do we want some padding in between these?

screenshot 2017-11-08 21 41 28

File: README.md [hunk 13]
```diff
@@ -1,121 +1,103 @@
...
+and we extend the core `@example` annotation
+to display compiled Sass/Nunjucks output
+and render sample components.
+We also provide a `@preview` annotation
+for displaying color-palettes, text and spacing sizes,
+and modular ratios, and a `@font` annotation
+for displaying font-specimens.
 
-To install the necessary Node dependencies, run ``yarn``.
+[See the full documentation for details »][docs]
```
Comment (jgerigmeyer) [inline README.md]: I think this should become a link to the annotation-specific docs (#183), and we should add a (recursive) link to these docs at the very top.

File: README.md (hunk 12 above)
Comment (jgerigmeyer) [reply README.md]: Weird, now it looks fine to me after being merged to `master`...

File: README.md (hunk 13 above)
Comment (mirisuzanne) [reply README.md]: Sounds good! We can do that in the new issue.

File: README.md (hunk 12 above)
Comment (mirisuzanne) [reply README.md]: Yeah, we could try adding `&nbsp;` between them?

**Pull request merged**